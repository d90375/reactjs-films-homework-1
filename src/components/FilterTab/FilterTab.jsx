import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from './FilterTab.scss';

const FilterTab = ({
  children, fetchByFilter, condition, filter,
}) => {
  const className = condition === children ? `${styles.tab} ${styles.active}` : styles.tab;
  return (
    <Link to={`/${filter}`}>
      <button
        type="button"
        className={className}
        data-filter={filter}
        onClick={(e) => fetchByFilter(e.target.dataset.filter)}
      >
        {children}
      </button>
    </Link>
  );
};

FilterTab.propTypes = {
  children: PropTypes.string.isRequired,
  fetchByFilter: PropTypes.func.isRequired,
  condition: PropTypes.string.isRequired,
  filter: PropTypes.string.isRequired,
};

export default FilterTab;
