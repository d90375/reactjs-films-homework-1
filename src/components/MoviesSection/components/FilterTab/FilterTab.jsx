import React from 'react';
import PropTypes from 'prop-types';

import styles from './FilterTab.scss';

const FilterTab = ({ children, fetchByFilter }) => (
  <p className={styles.tab} onClick={(e) => fetchByFilter(e.target.innerText)}>
    {children}
  </p>
);

FilterTab.propTypes = {
  children: PropTypes.string.isRequired,
  fetchByFilter: PropTypes.func.isRequired,
};

export default FilterTab;
