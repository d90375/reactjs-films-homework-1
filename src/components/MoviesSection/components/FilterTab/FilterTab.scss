.tab {
  padding: 0 10px;
  margin-bottom: 5px;
  line-height: 20px;
  font-weight: bold;
  color: #0aaee4;
  cursor: pointer;
}
