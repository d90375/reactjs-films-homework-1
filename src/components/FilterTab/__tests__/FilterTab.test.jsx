import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';
import FilterTab from '../FilterTab';

describe('FilterTab tests', () => {
  describe('FilterTab render', () => {
    it('FilterTab renders correctly', () => {
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<FilterTab fetchByFilter={mockCallBack} filter="Trending" condition="Trending">Trending</FilterTab>);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });

    it('FilterTab renders correctly when genres tab active', () => {
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<FilterTab fetchByFilter={mockCallBack} filter="Trending" condition="123">Trending</FilterTab>);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });

  describe('onClick function', () => {
    it('onClick callback function called', () => {
      const mockCallBack = jest.fn();

      const tree = create(
        <Router>
          <FilterTab
            fetchByFilter={mockCallBack}
            filter="Trending"
            condition="Trending"
            onClick={mockCallBack}
          >
            Trending
          </FilterTab>
        </Router>
        ,
      );

      const button = tree.root.findByProps({ type: 'button' });
      const e = {
        target: {
          dataset: {
            filter: 'Trending',
          },
        },
      };

      button.props.onClick(e);

      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack).toHaveBeenCalledWith('Trending');
    });
  });
});
