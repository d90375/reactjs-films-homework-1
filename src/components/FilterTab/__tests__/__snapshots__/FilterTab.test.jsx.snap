// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTab tests FilterTab render FilterTab renders correctly 1`] = `
<Link
  to="/Trending"
>
  <button
    className="tab active"
    data-filter="Trending"
    onClick={[Function]}
    type="button"
  >
    Trending
  </button>
</Link>
`;

exports[`FilterTab tests FilterTab render FilterTab renders correctly when genres tab active 1`] = `
<Link
  to="/Trending"
>
  <button
    className="tab"
    data-filter="Trending"
    onClick={[Function]}
    type="button"
  >
    Trending
  </button>
</Link>
`;
